File: 05. Sorting/01. Normal Sorting Algorithms/01. Bubble Sort/BubbleSort.java
import java.io.*;

class BubbleSort {

	static void swap(int[] arr, int i, int j) {
		int tmp = arr[i];
		arr[i] = arr[j];
		arr[j] = tmp;
	}

	static void bubbleSort(int[] arr, int n) {
		for (int i=0; i<n-1; i++) {
			boolean swapped = false;
			for (int j=0; j<n-i-1; j++) {
				if (arr[j] > arr[j+1]) {
					swap(arr, j, j+1);
					swapped = true;
				}
			}
			if (swapped == false) {break;}
		}
		for (int i=0; i<n; i++) {
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}

	public static void main(String[] args) throws IOException {
		int[] arr = {13, 32, 5, 0, 34, 67};
		bubbleSort(arr, arr.length);
	}
}

File: 05. Sorting/01. Normal Sorting Algorithms/01. Bubble Sort/BubbleSort.js
function swap(arr, i, j) {
  let tmp = arr[i];
  arr[i] = arr[j];
  arr[j] = tmp;
}

function bubbleSort(arr, n) {
  for (let i = 0; i < n-1; i++) {
    var swapped = false;
    for (let j=0; j<n-i-1; j++) {
      if (arr[j] > arr[j+1]) {
        swap(arr, j, j+1);
        swapped = true;
      }
    }
    if (swapped === false) {break;}
  }
}

var arr = [32,19,76,7,24];
bubbleSort(arr, arr.length);
var s = "";
for (let i=0; i<arr.length; i++) {
  s += arr[i]+" ";
}
console.log(s);
